File: Java/1603. Design Parking System.java
class ParkingSystem {
    public int parking[] = new int[]{0, 0, 0};

    public ParkingSystem(int big, int medium, int small) {
        parking[0] = big;
        parking[1] = medium;
        parking[2] = small;
    }
    
    public boolean addCar(int carType) {
        if(parking[carType - 1] == 0) return false;
        else parking[carType - 1]--;

        return true;
    }
}

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * ParkingSystem obj = new ParkingSystem(big, medium, small);
 * boolean param_1 = obj.addCar(carType);
 */

File: C++/1603. Design Parking System.cpp
class ParkingSystem {
public:
    int b, m, s;
    ParkingSystem(int big, int medium, int small) {
        b = big;
        m = medium;
        s = small;
    }
    
    bool addCar(int carType) {
        if(carType == 1){
            if(b == 0) return false;
            b--;
            return true;
        }
        else if(carType == 2){
            if(m == 0) return false;
            m--;
            return true;
        }
        else if(carType == 3){
            if(s == 0) return false;
            s--;
            return true;
        }
        
        return 0;
    }
};

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * ParkingSystem* obj = new ParkingSystem(big, medium, small);
 * bool param_1 = obj->addCar(carType);
 */
